client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -244,7 +245,7 @@ protected void checkBlockSendResult(Set<Long> blockIds) throws RuntimeException
 
   @VisibleForTesting
   protected void sendCommit() {
-    ExecutorService executor = Executors.newSingleThreadExecutor();
+    ExecutorService executor = Executors.newSingleThreadExecutor(ThreadUtils.getThreadFactory("sendCommit-%d"));
```

Comment:
This is a temporary thread pool. I think it's unnecessary to be a daemon thread pool.WDYT?